admin-dev/themes/default/template/controllers/orders/_shipping.tpl

```diff
@@ -54,40 +54,76 @@
 				<td class="weight">{$line.weight|string_format:"%.3f"} {Configuration::get('PS_WEIGHT_UNIT')}</td>
-				<td class="center">
+				<td class="price_carrier_{$line.id_carrier}" class="center">
+					<span>
 					{if $order->getTaxCalculationMethod() == $smarty.const.PS_TAX_INC}
 						{displayPrice price=$line.shipping_cost_tax_incl currency=$currency->id}
 					{else}
 						{displayPrice price=$line.shipping_cost_tax_excl currency=$currency->id}
 					{/if}
+					</span>
 				</td>
 				<td>
 					<span class="shipping_number_show">{if $line.url && $line.tracking_number}<a class="_blank" href="{$line.url|replace:'@':$line.tracking_number}">{$line.tracking_number}</a>{else}{$line.tracking_number}{/if}</span>
 				</td>
 				<td>
 					{if $line.can_edit}
-						<form method="post" action="{$link->getAdminLink('AdminOrders')|escape:'html':'UTF-8'}&amp;vieworder&amp;id_order={$order->id|intval}">
-							<span class="shipping_number_edit" style="display:none;">
-								<input type="hidden" name="id_order_carrier" value="{$line.id_order_carrier|htmlentities}" />
-								<input type="text" name="tracking_number" value="{$line.tracking_number|htmlentities}" />
-								<button type="submit" class="btn btn-default" name="submitShippingNumber">
-									<i class="icon-ok"></i>
-									{l s='Update'}
-								</button>
-							</span>
-							<a href="#" class="edit_shipping_number_link btn btn-default">
-								<i class="icon-pencil"></i>
-								{l s='Edit' d='Admin.Actions'}
-							</a>
-							<a href="#" class="cancel_shipping_number_link btn btn-default" style="display: none;">
-								<i class="icon-remove"></i>
-								{l s='Cancel' d='Admin.Actions'}
-							</a>
-						</form>
+						<a href="#" class="edit_shipping_link btn btn-default"
+						data-id-order-carrier="{$line.id_order_carrier|intval}"
+						data-id-carrier="{$line.id_carrier|intval}"
+						data-tracking-number="{$line.tracking_number|htmlentities}"
+						>
+ 							<i class="icon-pencil"></i>
+ 							{l s='Edit' d='Admin.Actions'}
+ 						</a>
 					{/if}
 				</td>
 			</tr>
 			{/foreach}
 		</tbody>
 	</table>
+
+	<!-- shipping update modal -->
+	<div class="modal fade" id="modal-shipping">
+		<div class="modal-dialog">
+			<form method="post" action="{$link->getAdminLink('AdminOrders')|escape:'html':'UTF-8'}&amp;vieworder&amp;id_order={$order->id|intval}">
+				<input type="hidden" name="submitShippingNumber" id="submitShippingNumber" value="1" />
+				<input type="hidden" name="id_order_carrier" id="id_order_carrier" />
+				<div class="modal-content">
+					<div class="modal-header">
+						<button type="button" class="close" data-dismiss="modal" aria-label="{l s='Close' d='Admin.Actions'}"><span aria-hidden="true">&times;</span></button>
+						<h4 class="modal-title">{l s='Edit shipping details'}</h4>
+					</div>
+					<div class="modal-body">
+						<div class="container-fluid">
+							{if !Configuration::get('PS_ORDER_RECALCULATE_SHIPPING')}
```

Comment:
Can you assign this first and then read the variable, instead of direct of using the Configuration class directly?
This breaks the MVC pattern and can have an impact on performance.